cmd/tendermint/commands/init.go

```diff
@@ -91,21 +105,43 @@ func initFilesWithConfig(config *cfg.Config) error {
 		ctx, cancel := context.WithTimeout(context.TODO(), ctxTimeout)
 		defer cancel()
 
-		pubKey, err := pv.GetPubKey(ctx)
-		if err != nil {
-			return fmt.Errorf("can't get pubkey: %w", err)
+		// if this is a validator we add it to genesis
+		if pv != nil {
+			pubKey, err := pv.GetPubKey(ctx)
+			if err != nil {
+				return fmt.Errorf("can't get pubkey: %w", err)
+			}
+			genDoc.Validators = []types.GenesisValidator{{
+				Address: pubKey.Address(),
+				PubKey:  pubKey,
+				Power:   10,
+			}}
 		}
-		genDoc.Validators = []types.GenesisValidator{{
-			Address: pubKey.Address(),
-			PubKey:  pubKey,
-			Power:   10,
-		}}
 
 		if err := genDoc.SaveAs(genFile); err != nil {
 			return err
 		}
 		logger.Info("Generated genesis file", "path", genFile)
 	}
 
+	// write config file
+	cfg.WriteConfigFile(config.RootDir, config)
+	logger.Info("Generated config", "mode", config.Mode)
+
 	return nil
 }
+
+func modePrompt() string {
+	fmt.Println("Please enter a node mode (validator|full|seed):")
+	scanner := bufio.NewScanner(os.Stdin)
+	for {
+		scanner.Scan()
+		response := scanner.Text()
+		switch response {
+		case cfg.ModeValidator, cfg.ModeFull, cfg.ModeSeed:
+			return response
+		default:
+			fmt.Println("Mode must be either validator, full or seed")
```

Comment:
shouldn't we use the logger? Are there ways that people run tendermint that could cause stdout to go somewhere different than the logger?

Reply:
I don't quite follow. Aren't we using a logger for this message?

Reply:
`fmt.Println` writes to `os.Stdout` always, and the logger could send output sort of anywhere? I think in common use fmt.Println goes to the same place, but its pretty common to send logs to other locations (e.g. syslog/stderr/other log aggregators), and in those cases the fmt.Printlns can end up getting lost or missed.

Reply:
Sorry yeah that makes sense. I've deleted this prompt function which somehow caused your comments to pop up on a different line. In this case, rather coincidentally, it was:
```
logger.Info("Generated config", "mode", config.Mode)
```
So I got a bit confused